src/Psalm/Internal/Visitor/CheckTrivialExprVisitor.php

```diff
@@ -0,0 +1,72 @@
+<?php
+namespace Psalm\Internal\Visitor;
+
+use PhpParser;
+
+/**
+ * @internal
+ */
+class CheckTrivialExprVisitor extends PhpParser\NodeVisitorAbstract implements PhpParser\NodeVisitor
+{
+    /**
+     * @var array<int, PhpParser\Node\Expr>
+     */
+    protected $non_trivial_expr = [];
+
+    /**
+     * @param  PhpParser\Node\Expr $node
+     * @return bool
+     */
+
+    private function checkNonTrivialExpr(PhpParser\Node\Expr $node)
+    {
+        if ($node instanceof PhpParser\Node\Expr\ArrayDimFetch ||
```

Comment:
I think this should be a whitelist of allowed expressions (variables, class constants) and not the other way around. That's avoid the issue where `PhpParser\Node\Expr\ArrowFunction` does not exist in some versions of the parser.